js/validation.js

```diff
@@ -0,0 +1,144 @@
+$(document).ready(function() {
+    var iserror = true;
+    $( 'form' ).on( 'submit', function() {
+        $('.empty').each(function() {
+            var error = '#'.concat($(this).attr('id'), '_err');
+
+            if (null === $(this).val() || '' === $(this).val()) {
+                $(error).text('This field is required.');
+                iserror = false;
+            }
+        });
+
+        return iserror;
+    });
+
+
+    $('.only-char').on('keyup blur paste', function() {
+        var letters = /^[a-zA-Z ]*$/;
+        var error = '#'.concat($(this).attr('id'), '_err');
+
+        if ( ! $(this).val().match(letters)) {
+            $(error).text('Only letters and white space allowed.');
+            iserror = false;
+        } else {
+            $(error).text('');
+            iserror = true;
+        }
+    });
+
+    $('.only-num').on('keyup blur paste', function() {
+        var number = /^[0-9 ]*$/;
+        var error = '#'.concat($(this).attr('id'), '_err');
+
+        if ( ! $(this).val().match(number)) {
+            $(error).text('Invalid data.');
+            iserror = false;
+        } else {
+            $(error).text('');
+            iserror = true;
+        }
+    });
+
+    $('.street').on('keyup blur paste', function() {
+        var street = /^[a-zA-Z\s\d-,]*$/;
+        var error = '#'.concat($(this).attr('id'), '_err');
+
+        if ( ! $(this).val().match(street)) {
+            $(error).text('Invalid data.');
+            iserror = false;
+        } else {
+            $(error).text('');
+            iserror = true;
+        }
+    });
+
+    $('#cpwd').on('keyup blur paste', function(){
+        var str_length = $(this).val().length;
+
+        if (str_length !== 0) {
+            if ($(this).val() !== $('#pwd').val().substr(0, str_length)) {
+                $('.cpwd_err').text('Password field does not match Confirm Password field');
+                iserror = false;
+            } else {
+                $('.cpwd_err').text('');
+                iserror = true;
+            }
+        }
+    });
+
+    $('.pin').on('blur', function(){
+        var str_length = $(this).val().length;
+        var error = '#'.concat($(this).attr('id'), '_err');
+
+        if (str_length !== 0 && str_length !== 6) {
+            $(error).text('Length must be 6.');
+        } else {
+            $(error).text('');;
+            iserror = true;
+        }
+    });
+
+    $('.phone').on('blur', function(){
+        var str_length = $(this).val().length;
+        var error = '#'.concat($(this).attr('id'), '_err');
+
+        if (str_length !== 0 && str_length !== 10) {
+            $(error).text('Length must be 10.');
+        } else {
+            $(error).text('');;
+            iserror = true;
+        }
+    });
+
+    $('.fax').on('blur', function(){
+        var str_length = $(this).val().length;
```

Comment:
Instead of using $(this) again, assign it to a variable like var $current_obj = $(this) and use $current_obj everywhere required. This is needed only if you are using $(this) more than once